cmd/snap/cmd_validate.go

```diff
@@ -0,0 +1,193 @@
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/snapcore/snapd/client"
+	"github.com/snapcore/snapd/i18n"
+)
+
+type cmdValidate struct {
+	clientMixin
+	Monitor bool `long:"monitor"`
+	// XXX: enforce mode is not supported yet
+	Enforce    bool `long:"enforce" hidden:"yes"`
+	Forget     bool `long:"forget"`
+	Positional struct {
+		ValidationSet string `positional-arg-name:"<validation-set>"`
+	} `positional-args:"yes"`
+	colorMixin
+}
+
+var shortValidateHelp = i18n.G("List or apply validation sets")
+var longValidateHelp = i18n.G(`
+The validate command lists or applies validations sets
+`)
+
+func init() {
+	addCommand("validate", shortValidateHelp, longValidateHelp, func() flags.Commander { return &cmdValidate{} }, colorDescs.also(map[string]string{
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"monitor": i18n.G("Monitor the given validations set"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"enforce": i18n.G("Enforce the given validation set"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		"forget": i18n.G("Forget the given validation set"),
+	}), []argDesc{{
+		// TRANSLATORS: This needs to begin with < and end with >
+		name: i18n.G("<validation-set>"),
+		// TRANSLATORS: This should not start with a lowercase letter.
+		desc: i18n.G("Validation set with an optional pinned sequence point, i.e. account/name[=seq]"),
+	}})
+}
+
+// this is reused for both account and set name of "account/name" argument
+var validName = regexp.MustCompile("^[a-z][0-9a-z]+$")
+
+func splitValidationSetArg(arg string) (account, name string, seq int, err error) {
+	parts := strings.Split(arg, "=")
+	if len(parts) > 2 {
+		return "", "", 0, fmt.Errorf("cannot parse validation set, expected account/name=seq")
+	}
+	if len(parts) == 2 {
+		seq, err = strconv.Atoi(parts[1])
+		if err != nil {
+			return "", "", 0, err
+		}
+	}
+
+	parts = strings.Split(parts[0], "/")
+	if len(parts) != 2 {
+		return "", "", 0, fmt.Errorf("expected a single account/name")
+	}
+
+	account = parts[0]
+	name = parts[1]
+	if !validName.MatchString(account) {
+		return "", "", 0, fmt.Errorf("invalid account name %q", account)
+	}
+	if !validName.MatchString(name) {
+		return "", "", 0, fmt.Errorf("invalid name %q", name)
+	}
+
+	return account, name, seq, nil
+}
+
+func fmtValid(res *client.ValidationSetResult) string {
+	if res.Valid {
+		return fmt.Sprintf("valid")
```

Comment:
Unless I miss something there is no need to use `fmt.Sprintf()` here, it's just a string without any formatting.